### DISCUSSION TITLE
Support for Windows Server 2022 (build 20348)

### DISCUSSION THREAD

**Discussion by kristibektashi** [valinet/ExplorerPatcher#992] [General]
Specifically for the weather widget as it just doesn't show up (I have already installed EdgeWebview2). Deselecting and reselecting it does nothing. Another interesting thing is that the Cortana button doesn't show up in Server 2022 even if it's selected in ExplorerPatcher (it just doesn't show up either, not that it would do anything but still)

**Comment by valinet** [comment]
Cortana doesn't show because it is either disabled by policy, or simply not included, so toggling the registry setting would obviously have no effect.

The weather widget *probably* doesn't work because the People bar is not available in Windows Server, so yeah, it's got nothing to replace.

**Reply by kristibektashi** [reply]
> Cortana doesn't show because it is either disabled by policy, or simply not included, so toggling the registry setting would obviously have no effect.

Cortana button shows if you run explorer.exe under the SYSTEM account (during the OOBE or by replacing sethc.exe with command prompt)

> The weather widget _probably_ doesn't work because the People bar is not available in Windows Server, so yeah, it's got nothing to replace.

Maybe you should use Windows Ink Workspace instead of People bar (or make it a toggleable option) since that exists on Windows Server. It's useless on Server anyway.

**Reply by kristibektashi** [reply]
Also cortana button sometimes shows for a split second when you set up a new user on Server 2022

**Reply by valinet** [reply]
> Not true

... I don't have that at hand. Obviously it doesn't work because something's missing from there...

> It's useless on Server anyway.

Same as People bar on consumer Windows. On my Windows 11 install I did not have Windows Ink showing, so I chose the People bar. There is always room for improvement, indeed.

**Comment by valinet** [comment]
Managed to get it working, will push an update soon. It's basically just the typical `IsOS(OS_ANYSERVER)` guarding.

WebView2 Runtime is installed automatically in the recent versions, that is if the people button is added there properly.

![image](https://user-images.githubusercontent.com/6503598/156832130-e9f4108d-f6c2-4e79-965e-a9f52ee8a805.png)

**Reply by valinet** [reply]
Enabled this in latest pre-release

**Comment by ideapc** [comment]
Hi, can you make EP Taskbar available for srv2022?

**Comment by kristibektashi** [comment]
Something I just realized: Perhaps the Cortana button WOULD show up if you were to actually install the (now-defunct) Cortana app for Windows 10 version 2004 and later, which of course is not included with Server 2022. Of course even if that is the case, this is pretty irrelevant but still noteworthy in my opinion.
